app/models/tariff_update.rb

```diff
@@ -5,52 +5,13 @@ class TariffUpdate
 
   collection_path '/updates/latest'
 
-  attr_accessor :update_type, :state, :created_at, :updated_at, :filename
-
-  def update_type
-    case attributes['update_type']
-    when /Taric/ then 'TARIC'
-    when /Chief/ then 'CHIEF'
-    end
-  end
-
-  def updated_at
-    date_attribute('updated_at')
-  end
-
   def applied_at
-    date_attribute('applied_at')
+    Date.parse(attributes['applied_at'].to_s)
   end
 
-  def to_s
-    "Applied #{update_type} at #{updated_at} (#{filename})"
-  end
-
-  class << self
-    def latest_applied_import_date
-      func = proc do
-        last = all.first
-        last.try(:applied_at) || Date.current
-      end
-
-      if Rails.env.test? || Rails.env.development?
-        # Do not cache it in Test and Development environments.
-        #
-        func.call
-      else
-        # Cache for 1 hour
-        TradeTariffFrontend::ServiceChooser.cache_with_service_choice(
-          'tariff_last_updated', expires_in: 1.hour
-        ) do
-          func.call
-        end
-      end
+  def self.latest_applied_import_date
+    TradeTariffFrontend::ServiceChooser.cache_with_service_choice('tariff_last_updated', expires_in: 23.hours) do
+      all.first.try(:applied_at) || Time.zone.today
```

Comment:
`first` should work as `all.first`, and it should be more efficient

Reply:
Not activerecord, sadly